photon-client/src/test/scala/com/linkedin/photon/ml/data/avro/AvroDataReaderTest.scala

```diff
@@ -106,6 +109,22 @@ class AvroDataReaderTest {
 
     AvroDataReader.readFeatureVectorFromRecord(record, Set(FeaturesField), indexMap)
   }
+
+  @Test
+  def testAllNumericTypes(): Unit = {
+    assertFalse(AvroDataReader.allNumericTypes(List(INT, STRING)))
+    assertTrue(AvroDataReader.allNumericTypes(List(INT, FLOAT)))
+    assertFalse(AvroDataReader.allNumericTypes(List(INT, LONG, FLOAT, DOUBLE, STRING)))
+    assertTrue(AvroDataReader.allNumericTypes(List(INT, LONG, FLOAT, DOUBLE)))
+  }
+
+  @Test
+  def testGetDominantNumericType(): Unit = {
+    assertEquals(AvroDataReader.getDominantNumericType(List(INT, LONG)), LONG)
+    assertEquals(AvroDataReader.getDominantNumericType(List(FLOAT, DOUBLE)), DOUBLE)
+    assertEquals(AvroDataReader.getDominantNumericType(List(INT, FLOAT)), FLOAT)
+    assertEquals(AvroDataReader.getDominantNumericType(List(INT, LONG, FLOAT, DOUBLE)), DOUBLE)
```

Comment:
I'd add one more test case where the dominant numeric type is in the middle of the list (right now it could pick the final item in the list each time and it would pass these tests).